### DISCUSSION TITLE
How Do I use Volo.Abp.Users.IUserLookupService

### DISCUSSION THREAD

**Discussion by alexkleinwaechter** [abpframework/abp#18053] [Q&A]
I try to inject IUserLookupService to my Appservice in order to enrich DTO's with user data like name and email.
I tried to follow instructions in this issue: https://github.com/abpframework/abp/issues/773
In version 7.4 the IUserLookupService needs apparently a UserType.
I tried it with:
 
`IUserLookupService<IUser> _userLookupService`

and 

`IUserLookupService<IdentityUser> _userLookupService`

But I get the followin error:
Cannot resolve parameter 'Volo.Abp.Users.IUserLookupService`1[Volo.Abp.Users.IUser] userLookupService'

Do I misunderstand something important?
It really should be easy to get some user context information in ABP.

Best!
Alex

**Comment by maliming** [comment]
> I try to inject IUserLookupService to my Appservice in order to enrich DTO's with user data like name and email.

You can get current user by `ICurrentUser` and get other users by the Identity module's services.

The `IUserLookupService` is not designed for your case.
